performant-software/dm-2
# Add Docker support, decouple from SendGrid, and improve docs (breaking change)

### What this PR does

- Per #399:
  - Adds Docker support in both dev and prod using Docker Compose
  - Decouples email from strictly SendGrid
  - Adds Heroku stack and other env variables to app.json, adds release phase to `Procfile`
  - Improves documentation in `README.md`
  - Adds Wiki page for upgrade to 1.0
  - Adds `.env.sample` and `application.sample.yml` files for environment variables

### Additional notes

Just before the 1.0 release, it may be wise to offload some of the more detailed installation instructions from the README, perhaps into the Wiki. This could improve organization and make it easier to find the right information.

#### Breaking change

In decoupling the app from SendGrid, environment variables have changed. `SENDGRID_USERNAME` and `SENDGRID_PASSWORD` have been renamed to `EMAIL_USERNAME` and `EMAIL_PASSWORD`, and additional variables `EMAIL_PORT` and `EMAIL_SERVER` have been added.

### Status

- [ ] Environment variables set in Heroku

### Steps to test
- Someone with Docker experience should test deploying the app in a development environment with Docker.

- Deploying the app in a prod environment is more involved, and we might have to wait for someone with a server already set up to serve from a domain name, and who has provisioned S3 and everything else necessary for production. Of course, if we have someone who can do this and is willing to test the Docker prod install, then that would be great.
